clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10079,6 +10079,28 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
         TCArgs.MakeArgString("--out-file-list=" + OutputFileName));
     ForeachArgs.push_back(
         TCArgs.MakeArgString("--out-replace=" + OutputFileName));
+    // If fsycl-dump-device-code is passed, put the output files from llvm-spirv
+    // into the path provided in fsycl-dump-device-code.
+    if (C.getDriver().isDumpDeviceCodeEnabled()) {
+      SmallString<128> OutputDir;
+
+      Arg *DumpDeviceCodeArg =
+          C.getArgs().getLastArg(options::OPT_fsycl_dump_device_code_EQ);
+
+      OutputDir = (DumpDeviceCodeArg ? DumpDeviceCodeArg->getValue() : "");
+
+      llvm::sys::path::remove_filename(OutputDir);
```

Comment:
`remove_filename` Removes the last component from path unless it is the root dir.
If the path is `/path/to/dir` , it will return `/path/to/`
If the path is `/path/to/dir/`, it will return `/path/to/dir/`

Reply:
We probably don't want to do this then.  If a user passes `/path/to/dir` they would expect the output to be in `/path/to/dir` not in `/path/to`